src/EnergyPlus/FileSystem.cc

```diff
@@ -210,25 +210,42 @@ namespace FileSystem {
         return fs::path{pext};
     }
 
-    // TODO: is this any better?
-    InputFileType getInputFileType(fs::path const &filePath)
+    FileTypes getFileType(fs::path const &filePath)
     {
-        std::string pext = fs::path(filePath).extension().string();
-        std::transform(pext.begin(), pext.end(), pext.begin(), ::toupper);
-        if (pext == ".EPJSON" || pext == ".JSON") {
-            return InputFileType::EpJSON;
-        } else if (pext == ".IDF" || pext == ".IMF") {
-            return InputFileType::IDF;
-        } else if (pext == ".CBOR") {
-            return InputFileType::CBOR;
-        } else if (pext == ".MSGPACK") {
-            return InputFileType::MsgPack;
-        } else if (pext == ".UBJSON") {
-            return InputFileType::UBJSON;
-        } else if (pext == ".BSON") {
-            return InputFileType::BSON;
+        std::string buffer{};
+        auto extension = std::string_view(fs::path(filePath).extension().c_str());
+        std::transform(extension.begin(), extension.end(), buffer.begin(), ::toupper);
+        auto const buffer_view = std::string_view(buffer.data());
+        if (buffer_view.compare(".EPJSON") == 0) {
```

Comment:
Should implement this using `getEnumerationValue` on a `constexpr std::array<std::string_view, FileTypes::Num>` that maps `FileTypes` enum to file extension strings.

Reply:
I'm not 100% convinced this is a good pattern for this use case but I implemented it anyways to test it out (I couldn't find any usage of `getEnumerationValue` in the codebase). My main worry is the enum and array getting out of sync but I think I took care of part of the problem with some `static_assert` statements. It does save me a string allocation though.

Reply:
That's fine, we'll change it later. The enumeration and the array getting out of sync is pearl clutching.

Reply:
I did say I implemented it right? 😉